src/python/analyzer_executor/src/analyzer_executor_lib/sqs_timeout_manager.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+import asyncio
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Coroutine
+
+from analyzer_executor_lib.sqs_types import SQSMessageId, SQSReceiptHandle
+from grapl_common.grapl_logger import get_module_grapl_logger
+
+if TYPE_CHECKING:
+    from mypy_boto3_sqs import SQSClient
+
+LOGGER = get_module_grapl_logger()
+
+
+class SqsTimeoutManagerException(Exception):
+    pass
+
+
+@dataclass
+class SqsTimeoutManager:
```

Comment:
Hello reviewer! I'd highly suggest, before you even look at the code of this class, to **read all the comments first.**

And heck, even read the comments on test_sqs_timeout_manager.py  !

Reply:
also compare with https://github.com/grapl-security/grapl/blob/staging/src/rust/sqs-executor/src/sqs_timeout_manager.rs